src/main/clojure/clara/rules/compiler.clj

```diff
@@ -235,15 +235,21 @@
   "Compiles a sequence of constraints into a structure that can be evaluated.
 
    Callers may also pass a collection of equality-only-variables, which instructs
-   this function to only do an equality check on them rather than create a unification binding."
+   this function to only do an equality check on them rather than create a unification binding.
+
+   The returned expression evaluates to a two item vector, [bindings error], which contains
+   the evaluated bindings in the first argument and a map with information on any exception
+   that might have been thrown by evaluating a constraint as the second vector entry."
```

Comment:
Hmm, bummer about the noise in the Clojurescript code. 

I do wonder if we could eliminate the try/catch per expression without losing much, and just have the try/catch in the alpha-node-matches in Will's example. So we'd let the exception thrown by an expression propagate to alpha-node-matches, and do the work there. We'd have the node that caused the exception there, and can get the full constraint from that node.

That should remove the ClojureScript dead code complaints and eliminate a bunch of generated try/catches, at the cost of some reduced granularity...but users will still see the exact node, high-level expression, and rules that caused it. 

Does this seem like it's worth considering?



 How much context do we lose if we just catch the in the outer expression